SETUP.md

```diff
@@ -36,7 +36,36 @@
       gcloud app deploy --project ${BBQ_PROJECT_ID} app.yaml config/cron.yaml config/queue.yaml config/index.yaml
       ```
       Note: If it is your first App Engine deploy, App Engine needs to be initialised and you will need to choose [region/location](https://cloud.google.com/appengine/docs/locations). It is recommended to pick the same location as where most of your BigQuery data resides.
-
+1.   Secure your application by following given steps.
+     * Restrict IAM roles for your GAE service account and Enable firewall for GAE application.
+       * GAE Editor needs to be removed manually, as it is set by default.
+         ```bash
+         gcloud projects remove-iam-policy-binding ${BBQ_PROJECT_ID} --member='serviceAccount:'${BBQ_PROJECT_ID}'@appspot.gserviceaccount.com' --role='roles/editor’
+         ```
+       * You should follow PoLP (Principle of least privilege) during process of set up IAM's.
+         You can do that via [Terraform](TERRAFORM_SETUP.md).
+         After setup go to **terraform/bbq** directory and run following command:
+         ```bash
+         terraform apply
+         ```
+         
+     * By default firewall blocked all traffic to your application.
+       To allow https traffic you need to whitelist your IP address into firewall rules.
+       You can do it via [UI](https://console.cloud.google.com/appengine/firewall) or run following command:
+       ```bash
+        gcloud app firewall-rules create 100 --action allow --source-range 0.0.0.0/0 --description 'my ip address'
+       ```
+       You can get your ip address with those commands:
+       * wireless
+       ```bash
+       ipconfig getifaddr en1
+       ``` 
+       * ethernet
+       ```bash
+       ipconfig getifaddr en0
+       ```
+     * Turn on [IAP](https://cloud.google.com/iap/docs/app-engine-quickstart) (Identity-Aware Proxy) for your GAE application.
```

Comment:
I think we should say something about when and how IAP should be configured

Reply:
👍